homeassistant/components/bmw_connected_drive/quality_scale.yaml

```diff
@@ -0,0 +1,71 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup:
+    status: exempt
+    comment: |
+      This integration doesn't have any events.
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions: done
+  config-entry-unloading: done
+  docs-configuration-parameters: done
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates: done
+  reauthentication-flow: done
+  test-coverage: done
+
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery-update-info:
+    status: exempt
+    comment: This integration doesn't use discovery.
+  discovery:
+    status: exempt
+    comment: This integration doesn't use discovery.
+  docs-data-update: todo
+  docs-examples: todo
+  docs-known-limitations: todo
+  docs-supported-devices: todo
+  docs-supported-functions: done
+  docs-troubleshooting: todo
+  docs-use-cases: todo
+  dynamic-devices: done
+  entity-category: done
+  entity-device-class: done
+  entity-disabled-by-default: done
+  entity-translations: done
+  exception-translations: todo
+  icon-translations: done
+  reconfiguration-flow: done
+  repair-issues:
+    status: exempt
+    comment: |
+      Other than reauthentication, this integration doesn't have any cases where raising an issue is needed.
+  stale-devices: done
+  # Platinum
+  async-dependency: done
+  inject-websession:
+    status: exempt
+    comment: The library requires a custom client for API authentication.
```

Comment:
I'm not sure about that one tbh. Sure, in the current implementation the library doesn't support it. But it's using `httpx`, so theoretically it would be possible (might need a custom `application_credentials` implementation though). Since this is a platinum criteria which is supposed to be the epitome, I'd vote for `todo`

Reply:
I agree here

Reply:
I'll put it to todo. However HA's `HassHttpXAsyncClient` will have a `httpx` User agent build custom in the helper. For BMW, we cannot use this as the API requires very specific user agents and even requires different user agents for different brands.
Not sure what happens if I overwrite this user agent - wouldn't this impact all HA httpx clients?

Reply:
You can also request a new client (instead of the shared one) from the helpers, maybe that’s a solution

Reply:
I'm marking this comment as resolved for now and changed the state to `todo` + a comment. Thanks for the input!

I will maybe try rebuilding/updating it after christmas, depending on the time I'll find.